deflate.c

```diff
@@ -1164,6 +1162,32 @@ Z_INTERNAL unsigned read_buf(PREFIX3(stream) *strm, unsigned char *buf, unsigned
     return len;
 }
 
+/* ===========================================================================
+ * Set longest match variables based on level configuration
+ */
+extern uint32_t update_hash_roll        (deflate_state *const s, uint32_t h, uint8_t val);
+extern uint32_t insert_string_roll      (deflate_state *const s, uint32_t str, uint32_t count);
+extern Pos      quick_insert_string_roll(deflate_state *const s, uint32_t str);
+
+static void lm_set_level(deflate_state *s, int level) {
+    s->max_lazy_match   = configuration_table[level].max_lazy;
+    s->good_match       = configuration_table[level].good_length;
+    s->nice_match       = configuration_table[level].nice_length;
+    s->max_chain_length = configuration_table[level].max_chain;
+
```

Comment:
This seems like a good place for a comment detailin the rationale of why update_hash/insert_string needs to be changed out.

Reply:
Done.